ibridgesgui/gui_utils.py

```diff
@@ -161,3 +166,41 @@ def get_downloads_dir() -> pathlib.Path:
     # Try to create Downloads
     pathlib.Path.home().joinpath("Downloads").mkdir(parents=True)
     return pathlib.Path.home().joinpath("Downloads")
+
+
+def get_tab_providers() -> list:
+    """Get a list of all environment template providers.
+
+    Returns
+    -------
+        The list that contains the providers.
+
+    """
+    return [entry.load() for entry in entry_points(group="ibridges_gui_tab")]
+
+def find_tab_provider(tab_providers: list, tab_name: str) -> object:
+    """Find the provider that provides the right template.
+
+    Parameters
+    ----------
+    tab_providers
+        A list of all installed tab providers.
+    tab_name
+        Name of the tab.
+
+    Returns
+    -------
+        The provider that contains the tab.
+
+    Raises
+    ------
+    ValueError
```

Comment:
Maybe KeyError makes more sense here?

Reply:
Maybe, I copied that function from the iBridges utils for the plugin there ... I would certainly like to keep them in sync.